content/rust/package/publish-deb-release-multi-arch.md

```diff
@@ -0,0 +1,468 @@
+$ cp target/debug/hello-world hello-world_1.0-1_amd64/bin
+```
+
+## Add Package Meta Data
+
+The next step is to create the necessary meta data file to describe the package.
+
+```
+$ mkdir hello-world_1.0-1_amd64/DEBIAN
+$ touch hello-world_1.0-1_amd64/DEBIAN/control
+```
+
+The `control` file is a simple text file which contains the meta data of the package.
+A complete documentation for this file can be found [on the official Debian site](https://www.debian.org/doc/debian-policy/ch-controlfields.html#binary-package-control-files-debian-control).
+For a simple package only a minimal content is needed:
+
+{{< highlight text "linenos=table,hl_lines=1-2 5-8" >}}
+Package: hello-world
+Version: 1.0-1
+Section: base
+Priority: optional
+Maintainer: Name <address@mail.com>
+Description: Hello World
+ A Drone tutorial application.
+Architecture: amd64
+
+{{< / highlight >}}
+
+{{< alert info >}}
+The last line needs to be empty.
+{{< / alert >}}
+
+The imported lines are highlighted.
+
+### The `Package` Attribute
+
+The `Package` attribute contains the name of the package.
+
+### The `Version` Attribute
+
+The `Version` attribute contains the application version of the package.
+
+### The `Maintainer` Attribute
+
+The `Maintainer` attribute should contain your name and email address.
+
+### The `Description` Attribute
+
+The `Description` attribute contains a short description of the application.
+If there are multiple lines they have to start with a space character.
+
+### The `Architecture` Attribute
+
+The `Architecture` attribute contains the platform where the package can be used.
+
+## Building the Debian Package
+
+The following tree shows the folder structure for the package:
+
+```
+hello-world_1.0-1_amd64
+|-> DEBIAN
+|---> control
+|-> bin
+|---> hello-world
+```
+
+To build the package [`dpkg-deb`](http://man7.org/linux/man-pages/man1/dpkg-deb.1.html) can be used:
+
+```console
+$ dpkg-deb --build hello-world_1.0-1_amd64
+dpkg-deb: building package 'hello-world' in 'hello-world_1.0-1_amd64.deb'.
+```
+
+After the build is finished a new file `hello-world_1.0-1_amd64.deb` is present in the current folder.
+The following screenshots show the package details:
+
+![Debian Package Info](/screenshots/rust-deb-package-info.png)
+
+To install the package [`dpkg`](http://man7.org/linux/man-pages/man1/dpkg.1.html) can be used:
+
+```console
+$ sudo dpkg -i hello-world_1.0-1_amd64.deb
+Selecting previously unselected package hello-world.
+(Reading database ... 185161 files and directories currently installed.)
+Preparing to unpack hello-world_1.0-1_amd64.deb ...
+Unpacking hello-world (1.0-1) ...
+Setting up hello-world (1.0-1) ...
+$ hello-world
+Hello, world!
+```
+
+# Simplify Package Creation
+
+The package creation should be automated to better fit into a continuous integration environment.
+A simple shell script (call it `pack.sh`) can combine all needed steps and can be transformed into a Drone plugin later.
+
```

Comment:
While I appreciate creating a plugin, I think this is outside the scope of this tutorial and we want to keep these tutorials focused. I think creating a custom plugin would be a great tutorial by itself.

For this tutorial can we invoke the commands directly in the `commands` section of the pipeline instead of creating a plugin?

Reply:
Sure, I just followed the DRY principle to not fill the yml with bloat.